
Author: aurelien
Status: I (rmh) reviewed manualy all the list, and removed lots of changes that
  were wrong or applied to non-existant packages.  Note: there's a section for
  adding Linux-specific packages at the end of the file.

Lazy Linux-specific add placeholder:  nfs-kernel-server

--- Packages-arch-specific.old	2005-10-29 22:45:45.000000000 +0200
+++ Packages-arch-specific	2005-10-29 23:26:05.000000000 +0200
@@ -1,7 +1,7 @@
 # Packages-arch-specific-version: 3
 # Quinn Diff packages-arch-specific file
 #
-# $Id: Packages-arch-specific,v 1.593 2005/10/12 06:14:53 adconrad Exp $
+# modifying this line in our patch helps us keep track of the original version
 #
 # Please email comments, corrections to james@nocrew.org and/or lamont@debian.org
 # See ChangeLog for history.
@@ -25,11 +25,11 @@
 855resolution: i386						      # i386 chipset specific
 aboot: alpha							      # alpha boot loader
 %aboot-installer: alpha					              # alpha boot loader installer	
-acpi: i386 ia64	amd64						      # acpi is i386/ia64 specific
+acpi: kfreebsd-i386 i386 ia64	amd64						      # acpi is i386/ia64 specific
 %acpi-support: i386 amd64 ia64					      # [ANAIS]
-acpid: i386 ia64 amd64						      # acpi is i386/ia64 specific
+acpid: kfreebsd-i386 i386 ia64 amd64						      # acpi is i386/ia64 specific
 %afbinit: sparc							      # Sparc gfx card firmware loader
-%alleyoop: i386 amd64						      # [ANAIS] - Depends valgrind
+%alleyoop: kfreebsd-i386 i386 amd64						      # [ANAIS] - Depends valgrind
 %alsadriver: i386                                                     # i386 specifc
 %alsalib: i386                                                        # i386 specifc
 alsautils: i386                                                       # i386 specifc
@@ -44,40 +44,40 @@
 atari-bootstrap: m68k                                                 # m68k specific
 atari-fdisk-cross: !m68k                                              # Everything but m68k
 atari-fdisk: m68k                                                     # m68k specific
-atitvout: i386							      # i386 specific
-%athcool: i386							      # Athlon powersave util
+atitvout: kfreebsd-i386 i386							      # i386 specific
+%athcool: kfreebsd-i386 i386							      # Athlon powersave util
 %autoinstall-i386: i386						      # i386 specific files
-%avifile: i386 amd64						      # i386 Win32 DLLs needed/executed
+%avifile: kfreebsd-i386 i386 amd64						      # i386 Win32 DLLs needed/executed
 um-pppd: hurd-i386						      # hurd
-%avrprog: i386 							      # inb/outb (ab)use
+%avrprog: kfreebsd-i386 i386 							      # inb/outb (ab)use
 %awe: i386 alpha                                                      # PC-Architecture specific
-%baycomepp: i386 alpha						      # [ANAIS]
+%baycomepp: kfreebsd-i386 i386 alpha						      # [ANAIS]
 %bigloo: !m68k							      # ROM
 binutils-m68k-linux: !m68k                                            # Is an m68k cross-compiler
 bluez-pcmcia-support: !s390					      # [ANAIS] depends: pcmcia-cs
-bmconf: i386							      # i386 boot loader
-brltty: alpha arm i386 sparc					      # [?] uses inb/outb
+bmconf: kfreebsd-i386 i386							      # i386 boot loader
+brltty: alpha arm kfreebsd-i386 i386 sparc					      # [?] uses inb/outb
 #bsign: !ia64 !alpha						      # ELF32 only
-bsign: i386							      # [ANAIS] see #148438
+bsign: kfreebsd-i386 i386							      # [ANAIS] see #148438
 %calltree: i386 amd64						      # Valgrind 'skin'
-%callgrind: i386 amd64                                                # depends on valgrind, #275851
-camediaplay: i386 alpha powerpc arm                                   # Attatches through the parallel port
+%callgrind: kfreebsd-i386 i386 amd64                                                # depends on valgrind, #275851
+camediaplay: kfreebsd-i386 i386 alpha powerpc arm                                   # Attatches through the parallel port
 catsboot: arm							      # arm bootloader
-%cbmlink: i386							      # sys/io.h
+%cbmlink: kfreebsd-i386 i386							      # sys/io.h
 cce: i386							      # i386 assembler
 %cfdisk-utf8: i386 arm hppa mipsel alpha ia64 powerpc mips	      # cfdisk is not used on all platforms
 cint: !m68k							      # ROM
-cmucl: i386							      # Yet to be ported to other arches
+cmucl: kfreebsd-i386 i386							      # Yet to be ported to other arches
 %cobalt-scripts: mipsel						      # [ANAIS]
 %colo: mipsel							      # mipsel boot loader
 %colo-installer: mipsel						      # mipsel boot loader
 corel-util: arm							      # arm specific
 %cpcieject: i386						      # Compaq/HP PCI Blade hot plug
-%cpuid: i386							      # x86 cpuid display program
+%cpuid: kfreebsd-i386 i386							      # x86 cpuid display program
 %cqcam: i386 alpha arm                                                # inb/outb (ab)use for port manipulation
-crash: i386 ia64 alpha powerpc					      # not yet ported to other platforms
+crash: kfreebsd-i386 i386 ia64 alpha powerpc					      # not yet ported to other platforms
 %ctypes: !m68k							      # ROM
-cpuburn: i386							      # i386 stress tester
+cpuburn: kfreebsd-i386 i386							      # i386 stress tester
 cwi-asfix-tools: !m68k                                                # maintainers choice
 cwi-aterm-tools: !m68k                                                # depends (eventually) on nfu cwi-* pkg
 cwi-graph-tools: !m68k                                                # depends (eventually) on nfu cwi-* pkg
@@ -92,17 +92,17 @@
 %cvsup: i386							      # i386 specifc (needs modula-3 )
 %detect: alpha i386						      # build-depend on isapnptools binary
 %delo-installer: mipsel						      # mipsel specific
-%dmidecode: i386 ia64 amd64					      # [ANAIS]
+%dmidecode: kfreebsd-i386 i386 ia64 amd64					      # [ANAIS]
 %drdsl: i386 amd64						      # [ANAIS]
-%drscheme: alpha amd64 hppa i386 m68k mips mipsel powerpc sparc	      # [ANAIS]
+%drscheme: alpha amd64 hppa kfreebsd-i386 i386 m68k mips mipsel powerpc sparc	      # [ANAIS]
 dosemu: i386                                                          # Hardcoded i386 assembler
-drip: i386							      # build-depends on avifile lib
+drip: kfreebsd-i386 i386							      # build-depends on avifile lib
 dtlk: i386                                                            # Driver for i386-only card
-dvr: i386							      # build-depends on avifile lib
-e3: i386							      # i386 assembly
-eciadsl: !m68k							      # [ANAIS]
-%eep24c: i386							      # [?] ANAIS, <sys/io.h>
-efibootmgr: i386 ia64						      # ia64 specific
+dvr: kfreebsd-i386 i386							      # build-depends on avifile lib
+e3: kfreebsd-i386 i386							      # i386 assembly
+eciadsl: !m68k !kfreebsd-i386							      # [ANAIS]
+%eep24c: kfreebsd-i386 i386							      # [?] ANAIS, <sys/io.h>
+efibootmgr: kfreebsd-i386 i386 ia64						      # ia64 specific
 %efi-reader: ia64						      # ia64 specific
 %egcs1.1: i386 m68k                                                   # compat libs
 egcs64: sparc							      # sparc specfic
@@ -110,20 +110,20 @@
 elilo: i386 ia64						      # ia64 specific boot-loader
 %elilo-installer: ia64						      # Installer for ia64 boot-loader
 emacs19: !powerpc                                                     # too old
-%etherboot: i386						      # i386 specific
-%extipl: i386							      # PC boot loader
+%etherboot: kfreebsd-i386 i386						      # i386 specific
+%extipl: kfreebsd-i386 i386							      # PC boot loader
 fai-kernels: i386						      # i386 specific
-fdflush: alpha i386                                                   # i386/alpha specific
+fdflush: alpha kfreebsd-i386 i386                                                   # i386/alpha specific
 %fenris: i386							      # requires porting
 %firebird: i386							      # requires porting
 %firebird2: i386 hurd-i386					      # requires porting
 %fnfx: i386							      # toshiba laptop specific
 %fpc: i386 sparc powerpc					      # self-hosting pascal compiler
-freej: i386							      # contains i386 assembler
+freej: kfreebsd-i386 i386							      # contains i386 assembler
 %ftape: i386 alpha						      # [?] ANAIS (!m68k anyway, others?)
 %ftape-tools: i386 alpha arm                                          # [?] ANAIS (!m68k anyway, others?)
 galeon-snapshot: !m68k                                                # depends on mozilla-snapshot
-%gatos: i386							      # no upstream !i386 support
+%gatos: kfreebsd-i386 i386							      # no upstream !i386 support
 gcardscheme: i386 alpha powerpc					      # PCMCIA frontend
 gcc272: i386 m68k                                                     # too old
 %gcc-2.95: !hppa !ia64 						      # hppa uses gcc-3.o only
@@ -131,52 +131,52 @@
 gcc-m68k-gnu: m68k						      # m68k specific (Hurd crosscompiler)
 gcc-i386-gnu: i386                                    # make-cross    # i386 specific (Hurd crosscompiler)
 gcc-m68k-linux: !m68k                                                 # Is an m68k cross-compiler
-gccchecker: i386                                                      # Yet to be ported to other arches
-%ggcov: i386							      # [ANAIS] endianness issues, etc.
+gccchecker: kfreebsd-i386 i386                                                      # Yet to be ported to other arches
+%ggcov: kfreebsd-i386 i386							      # [ANAIS] endianness issues, etc.
 %ghc4: i386							      # Requires itself to build itself
 %ghc5: i386							      # Requires itself to build itself
 %gibraltar-bootcd: i386						      # [?] ANAIS
 ginaccint: !m68k						      # depends on cint
 %gkrellm-i8k2: i386						      # Dell (i386) laptop krell
-%gkrellm-i8k: i386						      # Dell (i386) laptop krell
-%gkrellm-x86info: i386						      # x86 Mhz plugin
-%gkrellongrun: i386						      # Crusoe (x86) monitor for gkrellm
-%glide: i386 alpha ia64 amd64					      # i386+alpha+... specific
+%gkrellm-i8k: kfreebsd-i386 i386						      # Dell (i386) laptop krell
+%gkrellm-x86info: kfreebsd-i386 i386						      # x86 Mhz plugin
+%gkrellongrun: kfreebsd-i386 i386						      # Crusoe (x86) monitor for gkrellm
+%glide: kfreebsd-i386 i386 alpha ia64 amd64					      # i386+alpha+... specific
 gmgaclock: i386							      # i386 specific (asm + sys/perm.h)
-%gmod: i386                                                           # i386 specific
+%gmod: kfreebsd-i386 i386                                                           # i386 specific
 %gpmudmon-applet: powerpc					      # PMUD is powerpc APM
-%gnu-efi: i386 ia64						      # ia64 specific
-%gnumach: hurd-i386 i386					      # hurd kernel
+%gnu-efi: kfreebsd-i386 i386 ia64						      # ia64 specific
+%gnumach: hurd-i386 kfreebsd-i386 i386					      # hurd kernel
 %gnumach1: hurd-i386 						      # hurd kernel
-%gpart: i386 hurd-i386 ia64 alpha arm mipsel amd64		      # little endian specific
-gprolog: i386 mips mipsel sparc alpha powerpc			      # from source
-%grub: i386 hurd-i386 amd64                                           # i386 boot loader
+%gpart: kfreebsd-i386 i386 hurd-i386 ia64 alpha arm mipsel amd64		      # little endian specific
+gprolog: kfreebsd-i386 i386 mips mipsel sparc alpha powerpc			      # from source
+%grub: kfreebsd-i386 i386 hurd-i386 amd64                                           # i386 boot loader
 grub2: !hppa !ia64 m68k						      # bootloader
-grubconf: i386 hurd-i386 freebsd-i386 netbsd-i386		      # i386 boot loader
-grub-installer: i386 amd64 hurd-i386                                  # only useful if you have grub
+grubconf: i386 hurd-i386 kfreebsd-i386 netbsd-i386		      # i386 boot loader
+grub-installer: kfreebsd-i386 i386 amd64 hurd-i386                                  # only useful if you have grub
 gspy: i386							      # i386 assembler source ccvt.S
 %gtkpbbuttons: powerpc						      # powerpc specific
 %hostap-modules-i386: i386					      # i386 specific
 %hurd: hurd-i386						      # hurd kernel
-%kfreebsd-5: freebsd-i386					      # freebsd kernel
-hwtools: i386                                                         # Contains i386 specific binaries
+%kfreebsd-5: kfreebsd-i386					      # freebsd kernel
+hwtools: kfreebsd-i386 i386                                                         # Contains i386 specific binaries
 %i2c: i386							      # i386 specific
 %i2c-old: i386							      # [ANAIS] i386 specific
-%i810switch: i386						      # i386 specific
+%i810switch: kfreebsd-i386 i386						      # i386 specific
 %i855-crt: i386							      # [ANAIS] i386 specific
-%i8kutils: i386							      # Dell (i386) laptop utils
+%i8kutils: kfreebsd-i386 i386							      # Dell (i386) laptop utils
 %ibcs: i386 sparc				              	      # Yet to be ported to other arches
 %ikeyd: powerpc							      # iBook hotkey daemon
 intel-rng-tools: i386						      # i386 specific
 %ipmi-ctl: i386 amd64						      # i386 specific
 ipmitool: i386 amd64						      # i386 specific
-iroffer: i386 sparc powerpc alpha                                     # lacks config for other archs
+iroffer: kfreebsd-i386 i386 sparc powerpc alpha                                     # lacks config for other archs
 %isapnptools: i386 alpha arm                                          # ISA is PC architecture specific
 %jamvm: !m68k !mips !mipsel !sparc !hppa !ia64 !alpha !s390	      # [ANAIS]
-joystick: i386 arm alpha ia64                                         # others with analog joysticks?
-jslaunch: alpha arm i386 sparc					      # no inb on m68k
-%k6fftwgel: i386						      # fftw optimised for K6 CPU
-%k7fftwgel: i386						      # fftw optimised for K7 CPU
+joystick: kfreebsd-i386 i386 arm alpha ia64                                         # others with analog joysticks?
+jslaunch: alpha arm kfreebsd-i386 i386 sparc					      # no inb on m68k
+%k6fftwgel: kfreebsd-i386 i386						      # fftw optimised for K6 CPU
+%k7fftwgel: kfreebsd-i386 i386						      # fftw optimised for K7 CPU
 %kbd-chooser: !s390						      # s390 installs don't use a keyboard
 %kernel-headers-2.2.19-m68k: m68k
 %kernel-headers-2.2.20-m68k: m68k
@@ -480,15 +480,15 @@
 %kernel-patch-powerpc-2.6.14: powerpc
 %kernel-source-2.0.38: !m68k !powerpc !arm                            # 2.0.38 never released on m68k/powerpc
 %kernel-source-2.4.5-r4k-kn04: mipsel				      # mipsel kernel source
-kforth: i386							      # i386 only compiler
+kforth: kfreebsd-i386 i386							      # i386 only compiler
 kissme: i386							      # i386 assembler
-kon2: i386                                                            # Hardcoded i386 assembler
+kon2: kfreebsd-i386 i386                                                            # Hardcoded i386 assembler
 %ksetisaver: alpha hppa i386 ia64 powerpc sparc			      # #229606
 %ia32-libs: amd64 ia64						      # ia32 compat libs for amd64,ia64
 %ia32-libs-gtk: amd64 ia64					      # ia32 compat libs for amd64,ia64
 %ia32-libs-kde: amd64 ia64					      # ia32 compat libs for amd64,ia64
 %ia32-libs-openoffice.org: amd64 ia64				      # ia32 compat libs for amd64,ia64
-%ledstats: i386							      # [?] ANAIS, <sys/io.h>
+%ledstats: kfreebsd-i386 i386							      # [?] ANAIS, <sys/io.h>
 %libc-sparc: sparc                                                    # old libc for sparc
 %libdb: i386                                                          # db1 compatibility library
 %libdbd-informix-perl: i386                                           # ANAIS
@@ -498,11 +498,11 @@
 %libffm: alpha							      # alpha specific library
 %libg++27: i386							      # too old
 %libgcr410: i386						      # [ANAIS]
-%libjpeg-mmx: i386						      # mmx-enhanced libjpeg
+%libjpeg-mmx: kfreebsd-i386 i386						      # mmx-enhanced libjpeg
 %libmovtar: i386						      # i386 assembler
 %libsem: hurd-i386						      # hurd specific
 %libsendmail-milter-perl: !ia64					      # ANAIS
-%libparportled: i386						      # [?] ANAIS, <sys/io.h>
+%libparportled: kfreebsd-i386 i386						      # [?] ANAIS, <sys/io.h>
 %libpfm: ia64							      # library for pfmon
 %libpfm2: ia64							      # library for pfmon
 %libpfm3: ia64							      # library for pfmon
@@ -510,10 +510,10 @@
 %libxbox: i386                                                        # x86 xbox specific
 %lids-2.2: i386							      # [?] (see -devel archives)
 %lids-2.4: i386							      # [?] (see -devel archives)
-%lightning: i386 powerpc sparc					      # not supported upstream for others
+%lightning: kfreebsd-i386 i386 powerpc sparc					      # not supported upstream for others
 %lilo: i386 amd64                                                     # i386 boot loader
 lilo-installer: i386 amd64					      # only useful if you have lilo
-%linux32: !m68k !alpha					 	      # m68k is always 32-bit and alpha is always 64
+%linux32: !m68k !alpha !kfreebsd-i386					 	      # m68k is always 32-bit and alpha is always 64
 %linux-kernel-di: i386						      # i386 specific
 %linux-kernel-di-386: i386
 %linux-kernel-di-i386: i386
@@ -541,28 +541,28 @@
 %linux-kernel-di-s390-2.6: s390
 %linux-restricted-modules-2.6.10: i386 powerpc amd64		      # ANAIS
 %linux-wlan-ng: amd64 i386 powerpc arm alpha hppa		      # ANAIS [?]
-lirc-svga: i386							      # ANAIS, #267323
+lirc-svga: kfreebsd-i386 i386							      # ANAIS, #267323
 lkl: i386 amd64							      # i386 kb hardware specific
 %lm-sensors-old: i386						      # ANAIS
-lphdisk: i386							      # i386 laptop specific
+lphdisk: kfreebsd-i386 i386							      # i386 laptop specific
 lsadb: powerpc							      # macintosh-only bus
 %lsbdev: i386							      # ANAIS
 loadlin: i386                                                         # i386 boot loader
-longrun: i386							      # i386 Crusoe specific
-ltrace: i386 m68k arm powerpc s390 amd64                              # [p] needs ported
+longrun: kfreebsd-i386 i386							      # i386 Crusoe specific
+ltrace: kfreebsd-i386 i386 m68k arm powerpc s390 amd64                              # [p] needs ported
 %lush: !ia64							      # ANAIS, #267494
 %maxdb-7.5.00: i386 amd64 ia64					      # [ANAIS]
 %maxdb-buildtools: i386 amd64 ia64				      # [ANAIS]
 %mac-fdisk: m68k powerpc 					      # Apple Mac fdisk
 %m68kboot: m68k							      # m68k boot loader
-mbr: i386                                                             # i386 boot loader
+mbr: kfreebsd-i386 i386                                                             # i386 boot loader
 %mcelog: i386 amd64						      # [ANAIS]
 mdutils: !hppa !ia64						      # we don't do 2.2 kernels
-%memprof: i386							      # [?] upstream says i386 only
-memtest86: i386							      # i386 memory tester (bypasses OS entirely)
-%memtest86+: i386 amd64						      # i386 memory tester (bypasses OS entirely)
-%mga-vid: !m68k							      # [ANAIS]
-%mgapdesk: i386							      # [?]
+%memprof: kfreebsd-i386 i386							      # [?] upstream says i386 only
+memtest86: kfreebsd-i386 i386							      # i386 memory tester (bypasses OS entirely)
+%memtest86+: kfreebsd-i386 i386 amd64						      # i386 memory tester (bypasses OS entirely)
+%mga-vid: !m68k !kfreebsd-i386							      # [ANAIS]
+%mgapdesk: kfreebsd-i386 i386							      # [?]
 mig-m68k-gnu: m68k						      # kludge [stable only; removed from unstable]
 %mig: hurd-i386							      # hurd cross compiler
 mig-i386-gnu: i386                                                    # i386 specific
@@ -570,18 +570,18 @@
 %mindi-busybox: i386						      # [ANAIS] depends mindi
 %mindi-kernel: i386						      # [ANAIS] depends mindi
 %mindi-partimagehack: i386					      # [ANAIS] depends mindi
-mit-scheme: i386						      # Circular self-build-depends (requires itself to build itself)
+mit-scheme: kfreebsd-i386 i386						      # Circular self-build-depends (requires itself to build itself)
 mkinitrd-cd: i386						      # i386 specific
-mknbi: i386							      # i386 specific
-mkrboot: i386                                                         # i386 boot loader utilities
+mknbi: kfreebsd-i386 i386							      # i386 specific
+mkrboot: kfreebsd-i386 i386                                                         # i386 boot loader utilities
 %mkvmlinuz: powerpc						      # Powerpc kernel from open firmware.
-mlton: i386 hppa powerpc sparc					      # Self-hosting ML compiler
-mmx-emu: i386                                                         # i386 specific
+mlton: kfreebsd-i386 i386 hppa powerpc sparc					      # Self-hosting ML compiler
+mmx-emu: kfreebsd-i386 i386                                                         # i386 specific
 mocka: i386							      # i386 only compiler
 mondo: i386							      # needs mindi
 morse: i386							      # i386 specific
 mosixview: i386							      # depends on i386-only mosix
-%motioneye: i386						      # Sony VAIO (i386 laptop) specific
+%motioneye: kfreebsd-i386 i386						      # Sony VAIO (i386 laptop) specific
 %mosix: i386							      # i386 specific currently
 %mozart: !alpha !ia64						      # 'who would ever need > 32-bits??'
 %mozart-gtk: !alpha !ia64					      # depends mozart
@@ -592,15 +592,15 @@
 %ncurses4.2: alpha arm i386 m68k powerpc sparc                        # compat only
 %ndiswrapper: i386 amd64                                              # Windows DLL loader
 %ndiswrapper-modules-i386: i386 amd64                                 # Windows DLL loader
-%neon0.23: i386 powerpc s390 sparc amd64			      # [ANAIS] revived only for OOo, which only exists there
+%neon0.23: kfreebsd-i386 i386 powerpc s390 sparc amd64			      # [ANAIS] revived only for OOo, which only exists there
 netcfg-dhcp: !s390						      # s390 installs don't use DHCP
 %newlib: !hppa !ia64						      # [ANAIS]
 %nhc98: !ia64 !alpha !amd64 !sparc !m68k			      # Per Ian Lynagh <igloo@earth.li>
-%nictools-nopci: i386						      # [?] ISA nictools
-%nictools-pci: i386 arm alpha					      # [?] PCI nictools
+%nictools-nopci: kfreebsd-i386 i386						      # [?] ISA nictools
+%nictools-pci: kfreebsd-i386 i386 arm alpha					      # [?] PCI nictools
 nvram: m68k                                                           # m68k specific
 nvram-wakeup: i386						      # i386 BIOS specific
-%nvtv: i386							      # [?] ANAIS, <sys/io.h>
+%nvtv: kfreebsd-i386 i386							      # [?] ANAIS, <sys/io.h>
 nwutil: arm							      # arm only
 %nw802: i386							      # [?] ANAIS, <asm/io.h>
 %ocamlgsl: !hppa !sparc						      # [ANAIS] upstream and alignment issues
@@ -612,11 +612,11 @@
 %openmcl-build-tools: powerpc					      # powerpc only assembly
 %openmosix: i386						      # i386 specific currently
 %openmosixview: i386						      # depends on i386-only openmosix
-%oprofile: amd64 i386 ia64 alpha hppa powerpc sparc		      # [ANAIS]
+%oprofile: amd64 kfreebsd-i386 i386 ia64 alpha hppa powerpc sparc		      # [ANAIS]
 oprofile-source: !m68k						      # [ANAIS] (others?)
 orp: i386							      # Intel's ix86 specific JVM
-oskit: i386 hurd-i386						      # i386 OS functions
-%p4fftwgel: i386						      # Pentium 4 specifc fftw lib
+oskit: kfreebsd-i386 i386 hurd-i386						      # i386 OS functions
+%p4fftwgel: kfreebsd-i386 i386						      # Pentium 4 specifc fftw lib
 %partimage: !ia64 !alpha !amd64					      # 64-bit is br0ken
 %paxtest: i386							      # [ANAIS]
 %pbbuttonsd: powerpc						      # powerpc specific
@@ -632,19 +632,19 @@
 pbuilder-uml: i386						      # depends on user-mode-linux
 pciutils: !m68k                                                       # m68k doesn't do PCI; others?
 #%pcmcia-cs: i386 arm alpha powerpc ia64                              # should build anywhere with PCI now?
-%pcmcia-cs: !s390						      # everywhere else has bridges.
+%pcmcia-cs: !s390 !kfreebsd-i386						      # everywhere else has bridges.
 %pcmcia-modules-2.4.26-i386: i386				      # i386 specific
 %pcmcia-modules-2.4.27-i386: i386				      # i386 specific
 %pcmcia-modules-2.4.28-i386: i386				      # i386 specific
 pearpc: i386 amd64 hurd-i386					      # [ANAIS]
 penguin: m68k                                                         # m68k boot loader
 %pfmon: ia64							      # ia64 specific
-pforth: arm i386 m68k powerpc s390 sparc hurd-i386 hppa		      # requires explicit porting
+pforth: arm kfreebsd-i386 i386 m68k powerpc s390 sparc hurd-i386 hppa		      # requires explicit porting
 %pgi: i386 ia64 powerpc						      # Installer -- needs porting work
-#photopc: alpha arm hurd-i386 i386 powerpc sparc			      # Attatches through the parallel port - removed per tb@becket.net
+#photopc: alpha arm hurd-i386 kfreebsd-i386 i386 powerpc sparc			      # Attatches through the parallel port - removed per tb@becket.net
 %php4-interbase: i386						      # depends on firebird
 picturebook: i386						      # i386 specific
-pileup: i386 alpha                                                    # inb()/outb() (ab)use
+pileup: kfreebsd-i386 i386 alpha                                                    # inb()/outb() (ab)use
 %plex86: i386							      # i386 emulator
 %pm3: i386							      # Modula compiler for i386
 %pm3i386: i386							      # Modula compiler for i386 bootstrap
@@ -654,116 +654,116 @@
 powerpc-utils: powerpc						      # powerpc specific
 %prc-tools: !hppa !ia64						      # requires gcc 2.95
 %prctl: hppa ia64						      # Only hppa/ia64 have syscalls for it
-%prelink: alpha amd64 i386 powerpc				      # No upstream support for others
-primaxscan: i386						      # [?]
+%prelink: alpha amd64 kfreebsd-i386 i386 powerpc				      # No upstream support for others
+primaxscan: kfreebsd-i386 i386						      # [?]
 %proll: sparc							      # Build-deps sparc-utils
 %prospect: i386 ia64						      # [ANAIS]
 pstack: i386							      # needs knowledge of stack layout
-%psyco: i386							      # [ANAIS] no upstream support
-%python-kinterbasdb: i386					      # depends on firebird
+%psyco: kfreebsd-i386 i386							      # [ANAIS] no upstream support
+%python-kinterbasdb: kfreebsd-i386 i386					      # depends on firebird
 %pupa: i386							      # i386 specific boot loader
 %qemu: !hppa !ia64 !mipsel					      # [ANAIS] upstream supports all of debian's archs except these.
 %quadra: i386							      # lots of portability issues
 q-tools: ia64							      # ia64 perf utils
-qcam: alpha i386 arm						      # Attaches through the parallel port
+qcam: alpha kfreebsd-i386 i386 arm						      # Attaches through the parallel port
 quik: powerpc							      # powerpc boot loader
 quik-installer: powerpc						      # powerpc boot loader
 qvplay: i386                                                          # Attatches through the parallel port
 raidtools: !ia64 !hppa
-%read-edid: i386	 					      # ioperm()/iopl(), vm86() etc. i386 specific
+%read-edid: kfreebsd-i386 i386	 					      # ioperm()/iopl(), vm86() etc. i386 specific
 red-carpet: i386						      # brain damaged upstream...
-replicator: i386						      # i386 specific
-rio: i386 alpha							      # inb/outb (ab)use for port manipulation
+replicator: kfreebsd-i386 i386						      # i386 specific
+rio: kfreebsd-i386 i386 alpha							      # inb/outb (ab)use for port manipulation
 %rootstrap: i386						      # Depends on user-mode-linux
 %rsbac-admin: i386 sparc powerpc				      # [ANAIS] #229416
-%rscheme: i386 hurd-i386 powerpc alpha m68k s390		      # [p] Needs per-arch MM glue (cr2)
-%rscheme-modules: i386 hurd-i386 powerpc alpha s390		      # [p] depends on rscheme (& m68k has no fshell)
+%rscheme: kfreebsd-i386 i386 hurd-i386 powerpc alpha m68k s390		      # [p] Needs per-arch MM glue (cr2)
+%rscheme-modules: kfreebsd-i386 i386 hurd-i386 powerpc alpha s390		      # [p] depends on rscheme (& m68k has no fshell)
 %rsjog: i386							      # Sony Vaio Jog Button program
 %rtai: i386 powerpc mips mipsel arm				      # [ANAIS]
 rtlinux: i386							      # currently i386 specific
 %s390-dasd: s390						      # s390 hard disk configurator
 %s390-netdevice: s390						      # s390 network configurator
 %s390-tools: s390						      # s390 specific utilities
-%s3switch: i386							      # i386 laptop gfx card prog
+%s3switch: kfreebsd-i386 i386							      # i386 laptop gfx card prog
 %salinfo: ia64							      # ia64 firmware monitor/decoder
-%sbcl: i386 powerpc alpha sparc mips mipsel			      # cmucl fork (asm, needs port)
+%sbcl: kfreebsd-i386 i386 powerpc alpha sparc mips mipsel			      # cmucl fork (asm, needs port)
 %sbcl-mt: i386 							      # build-depends sbcl
-sbm: i386							      # i386 boot loader
+sbm: kfreebsd-i386 i386							      # i386 boot loader
 sccw: i386                                                            # Relies on PC Sound hardware
 %sced: !alpha !ia64						      # Depends on elk
 %sdl-stretch: i386 amd64 hurd-i386				      # [ANAIS]
 sensor-sweep-applet: i386					      # i386 specific
-set6x86: i386                                                         # i386 specific
+set6x86: kfreebsd-i386 i386                                                         # i386 specific
 setsccserial: m68k                                                    # m68k specific
 sibyl: mips mipsel						      # mips, mipsel boot loader
 sibyl-installer: mips mipsel					      # mips, mipsel boot loader
 silo: sparc							      # sparc lilo
 %silo-installer: sparc						      # sparc lilo
-%sjog: i386							      # Sony VAIO laptop tool
+%sjog: kfreebsd-i386 i386							      # Sony VAIO laptop tool
 %skribe: !m68k							      # build-depends on bigloo
-%sleepd: alpha arm i386 m68k powerpc sparc			      # [?] APM sleep daemon
+%sleepd: alpha arm kfreebsd-i386 i386 m68k powerpc sparc			      # [?] APM sleep daemon
 sml-nj: i386                                                          # i386 specific
 %sndconfig: i386						      # depends on kudzu and isapnptools
 sparc-utils: sparc                                                    # sparc specific
 speedtouch: !m68k						      # [ANAIS]
 %spicctrl: i386							      # For Sony VAIO laptops
-%splay: i386 alpha arm m68k powerpc				      # [?] broken on sparc (see BTS]
+%splay: kfreebsd-i386 i386 alpha arm m68k powerpc				      # [?] broken on sparc (see BTS]
 %sppc: !m68k							      # no xlibosmesa-dev build-dep for m68k
 %srm-reader: alpha						      # Alpha specific udeb
-%stalin: alpha arm i386 sparc					      # [ANAIS]
+%stalin: alpha arm kfreebsd-i386 i386 sparc					      # [ANAIS]
 %strategoxt: i386 s390 sparc powerpc hppa mipsel mips		      # b-ds on xt-sdf2 package, doesn't work on arm #230574
-subterfugue: i386						      # i386 specific process tracing etc.
+subterfugue: kfreebsd-i386 i386						      # i386 specific process tracing etc.
 %supercollider: !alpha !ia64 !amd64				      # doesn't work with 64bit, #276212
-%svgalib: i386                                                        # PC-Architecture specific
-svgatextmode: i386                                                    # Hardcoded i386 assembler
-syslinux: i386 amd64                                                  # i386 boot loader thingy
-%systemimager: i386						      # hand built, to be truthful.
+%svgalib: kfreebsd-i386 i386                                                        # PC-Architecture specific
+svgatextmode: kfreebsd-i386 i386                                                    # Hardcoded i386 assembler
+syslinux: kfreebsd-i386 i386 amd64                                                  # i386 boot loader thingy
+%systemimager: kfreebsd-i386 i386						      # hand built, to be truthful.
 %systrace: i386							      # Needs porting
 task-laptop: i386 powerpc					      # PCMCIA specific
-tcc: i386							      # i386 only C compiler.
+tcc: kfreebsd-i386 i386							      # i386 only C compiler.
 %tempest: !s390							      # Needs xserver-xfree86, #229612
-tendra: i386							      # i386 specific (compiler)
+tendra: kfreebsd-i386 i386							      # i386 specific (compiler)
 termcap-compat: i386						      # needed only on i386
 %thinkpad: i386							      # i386 only laptop
 %tom: alpha i386 m68k powerpc					      # [?] needs porting
-toshset: i386							      # i386 specific
-toshutils: i386                                                       # i386 specific
-%tpb: i386							      # i386 specific
+toshset: kfreebsd-i386 i386							      # i386 specific
+toshutils: kfreebsd-i386 i386                                                       # i386 specific
+%tpb: kfreebsd-i386 i386							      # i386 specific
 %tpctl: i386							      # i386 specific
-%tra: i386							      # [ANAIS]
-%tvtime: i386 amd64						      # i386 (wine) specific
-%u++: i386 ia64 amd64						      # per rcbilson@plg.uwaterloo.ca
+%tra: kfreebsd-i386 i386							      # [ANAIS]
+%tvtime: kfreebsd-i386 i386 amd64						      # i386 (wine) specific
+%u++: kfreebsd-i386 i386 ia64 amd64						      # per rcbilson@plg.uwaterloo.ca
 %uclibc: i386 powerpc mips mipsel sh m68k arm sparc		      # cut down libc, needs ported
 %umlrun: i386							      # Depends on user-mod-linux
 %user-mode-linux: i386						      # [?] ANAIS
-%valgrind: i386 amd64						      # Needs ported (asm, arch knwldge)
-%vbetool: i386                                                        # Uses vm86 to access x86 BIOS calls
+%valgrind: kfreebsd-i386 i386 amd64						      # Needs ported (asm, arch knwldge)
+%vbetool: kfreebsd-i386 i386                                                        # Uses vm86 to access x86 BIOS calls
 vcr: i386							      # build-depends on avifile lib
-%vcs-tree: i386 sparc powerpc mips mipsel alpha			      # [ANAIS]
+%vcs-tree: kfreebsd-i386 i386 sparc powerpc mips mipsel alpha			      # [ANAIS]
 %vegastrike: !m68k						      # requested by Mike Furr <mfurr@debian.org>, see 207578
-vgrabbj: i386							      # i386 assembler
+vgrabbj: kfreebsd-i386 i386							      # i386 assembler
 vmelilo: m68k							      # m68k (VME) lilo
 %vmelilo-installer: m68k					      # m68k (VME) lilo
 wanpipe: i386                                                         # i386-specific
 wdsetup: i386							      # i386 specific
-webmin-grub: i386 hurd-i386					      # Depends on grub
-%webmin-extra: i386						      # [ANAIS]
-welcome2l: i386 alpha						      # requires VGA hi-ascii characters
+webmin-grub: kfreebsd-i386 i386 hurd-i386					      # Depends on grub
+%webmin-extra: kfreebsd-i386 i386						      # [ANAIS]
+welcome2l: kfreebsd-i386 i386 alpha						      # requires VGA hi-ascii characters
 %widelands: !m68k						      # requires porting work
-%wine: i386                                                           # Hardcoded i386 assembler
-%winelib: i386							      # depends on wine
-winesetuptk: i386						      # depends on wine
-wmacpi: i386							      # i386 specific
+%wine: kfreebsd-i386 i386                                                           # Hardcoded i386 assembler
+%winelib: kfreebsd-i386 i386							      # depends on wine
+winesetuptk: kfreebsd-i386 i386						      # depends on wine
+wmacpi: kfreebsd-i386 i386							      # i386 specific
 wmbatppc: powerpc						      # powerpc battery monitor
 %wmtune: !hppa !ia64 !m68k					      # sys/io.h
-%x86info: i386							      # i386 CPU info prog
-xbox-raincoat: i386						      # i386 xbox BIOS flasher
+%x86info: kfreebsd-i386 i386							      # i386 CPU info prog
+xbox-raincoat: kfreebsd-i386 i386						      # i386 xbox BIOS flasher
 %xdslusb: i386							      # [ANAIS]
-%xen: i386							      # [ANAIS]
+%xen: kfreebsd-i386 i386							      # [ANAIS]
 %xfree86v3: alpha i386						      # i386 alpha need old servers
 %xfree86-driver-synaptics: !s390				      # Needs xserver-xfree86
 %xine: i386							      # i386 assembler
-%xmbmon: i386 hurd-i386 amd64 kfreebsd-i386			      # [ANAIS] sys/io.h
+%xmbmon: kfreebsd-i386 i386 hurd-i386 amd64 kfreebsd-i386			      # [ANAIS] sys/io.h
 %xmovie: i386							      # [ANAIS] i386 assembly
 xpmumon: powerpc						      # powerpc specific laptop power management
 xserver-8514: i386 hurd-i386                          # xfree86       # i386 specific
@@ -785,13 +785,13 @@
 xserver-tga: alpha				      # xfree86	      # alpha specific
 xserver-xfree86: !s390				      # xfree86       # s390 has no xserver, #229612
 %xsystrace: i386						      # Depends on systrace
-%xt-sdf2: i386 s390 sparc powerpc hppa arm mipsel mips		      # [ANAIS]
-%xt-toolbuslib: i386 s390 sparc powerpc hppa arm mipsel mips m68k     # [ANAIS] cwi-aterm no plans to fix upstream
+%xt-sdf2: kfreebsd-i386 i386 s390 sparc powerpc hppa arm mipsel mips		      # [ANAIS]
+%xt-toolbuslib: kfreebsd-i386 i386 s390 sparc powerpc hppa arm mipsel mips m68k     # [ANAIS] cwi-aterm no plans to fix upstream
 yaboot: powerpc							      # PowerPC Bootloader
 yaboot-installer: powerpc					      # PowerPC Bootloader installer
-yforth: i386 m68k sparc arm powerpc				      # compiler
+yforth: kfreebsd-i386 i386 m68k sparc arm powerpc				      # compiler
 z8530-utils: i386                                                     # PC-architecture specific
-zblast-svgalib: i386						      # i386 only per David Pashley
+zblast-svgalib: kfreebsd-i386 i386						      # i386 only per David Pashley
 zorroutils: m68k powerpc					      # zorro (Amiga) specific
 %zipl-installer: s390						      # installer stuff
 
@@ -828,75 +828,75 @@
 %xpm: i386							      # libc5 compat version of xpm
 
 # isdnutils has one arch: any package (isdnvboxclient), so we can't % it ....
-isdnutils: alpha arm i386 ia64 m68k powerpc sparc mips mipsel
-isdnutils-xtools: alpha arm i386 ia64 m68k powerpc sparc mips mipsel
-ipppd: alpha arm i386 ia64 m68k powerpc sparc mips mipsel
-isdnlog: alpha arm i386 ia64 m68k powerpc sparc mips mipsel
-isdnvbox: alpha arm i386 ia64 m68k powerpc sparc mips mipsel
-isdnvboxserver: alpha arm i386 ia64 m68k powerpc sparc mips mipsel
-libcapi20: alpha arm i386 ia64 m68k powerpc sparc mips mipsel
-isdnactivecards: alpha arm i386 ia64 m68k powerpc sparc mips mipsel
-pppdcapiplugin: alpha arm i386 ia64 m68k powerpc sparc mips mipsel
-isdneurofile: alpha arm i386 ia64 m68k powerpc sparc mips mipsel
-%chan-capi: alpha arm i386 ia64 m68k powerpc sparc mips mipsel
-%capi4hylafax: alpha arm i386 ia64 m68k powerpc sparc mips mipsel
+isdnutils: alpha arm kfreebsd-i386 i386 ia64 m68k powerpc sparc mips mipsel
+isdnutils-xtools: alpha arm kfreebsd-i386 i386 ia64 m68k powerpc sparc mips mipsel
+ipppd: alpha arm kfreebsd-i386 i386 ia64 m68k powerpc sparc mips mipsel
+isdnlog: alpha arm kfreebsd-i386 i386 ia64 m68k powerpc sparc mips mipsel
+isdnvbox: alpha arm kfreebsd-i386 i386 ia64 m68k powerpc sparc mips mipsel
+isdnvboxserver: alpha arm kfreebsd-i386 i386 ia64 m68k powerpc sparc mips mipsel
+libcapi20: alpha arm kfreebsd-i386 i386 ia64 m68k powerpc sparc mips mipsel
+isdnactivecards: alpha arm kfreebsd-i386 i386 ia64 m68k powerpc sparc mips mipsel
+pppdcapiplugin: alpha arm kfreebsd-i386 i386 ia64 m68k powerpc sparc mips mipsel
+isdneurofile: alpha arm kfreebsd-i386 i386 ia64 m68k powerpc sparc mips mipsel
+%chan-capi: alpha arm kfreebsd-i386 i386 ia64 m68k powerpc sparc mips mipsel
+%capi4hylafax: alpha arm kfreebsd-i386 i386 ia64 m68k powerpc sparc mips mipsel
 
 ## Depend: on svglaib
-acidwarp: i386
-argante: i386
-bmv: i386
-chdrv: i386
-dvisvga: i386
-lincity-svga: i386
-lockvc: i386
-luxman: i386
-razzle: i386
-sabre: i386
-svncviewer: i386
-vgacardgames: i386
-vgagamespack: i386
-yh: i386
-zgv: i386
+acidwarp: kfreebsd-i386 i386
+argante: kfreebsd-i386 i386
+bmv: kfreebsd-i386 i386
+chdrv: kfreebsd-i386 i386
+dvisvga: kfreebsd-i386 i386
+lincity-svga: kfreebsd-i386 i386
+lockvc: kfreebsd-i386 i386
+luxman: kfreebsd-i386 i386
+razzle: kfreebsd-i386 i386
+sabre: kfreebsd-i386 i386
+svncviewer: kfreebsd-i386 i386
+vgacardgames: kfreebsd-i386 i386
+vgagamespack: kfreebsd-i386 i386
+yh: kfreebsd-i386 i386
+zgv: kfreebsd-i386 i386
 
 ## Depend: on cmucl
 cil: i386
-%cmucl: i386
+%cmucl: kfreebsd-i386 i386
 series: i386
-%clue: i386
+%clue: kfreebsd-i386 i386
 
 ## Depend: on gnat
-%gnat: i386 sparc powerpc				      # [ANAIS] upstream support list
-%adasockets: i386 sparc powerpc
-%adabrowse: i386 sparc
-%adacgi: i386 sparc powerpc
-%asis: i386 sparc powerpc
-%gch: i386 sparc powerpc
+%gnat: kfreebsd-i386 i386 sparc powerpc				      # [ANAIS] upstream support list
+%adasockets: kfreebsd-i386 i386 sparc powerpc
+%adabrowse: kfreebsd-i386 i386 sparc
+%adacgi: kfreebsd-i386 i386 sparc powerpc
+%asis: kfreebsd-i386 i386 sparc powerpc
+%gch: kfreebsd-i386 i386 sparc powerpc
 %gnade: !arm !m68k
-%gnat-glade: i386 sparc powerpc
-gnat-gps: i386 sparc powerpc
+%gnat-glade: kfreebsd-i386 i386 sparc powerpc
+gnat-gps: kfreebsd-i386 i386 sparc powerpc
 %gvd: !arm !m68k
-%libadabindx: i386 sparc powerpc
-%libaunit: i386 sparc powerpc
-%libaws: i386 sparc powerpc				      # b-d on libxmlada1(-dev)
-%libcharles0: i386 sparc powerp
-%libflorist-3.15p-1: i386 sparc powerpc
-%libgtkada: i386 sparc powerpc
-%libgtkada2: i386 sparc powerpc
-%libopentoken: i386 sparc powerpc
-%libtexttools2: i386 sparc powerpc
-%libxmlada1: i386 sparc powerpc
-%topal: i386 sparc powerpc
+%libadabindx: kfreebsd-i386 i386 sparc powerpc
+%libaunit: kfreebsd-i386 i386 sparc powerpc
+%libaws: kfreebsd-i386 i386 sparc powerpc				      # b-d on libxmlada1(-dev)
+%libcharles0: kfreebsd-i386 i386 sparc powerp
+%libflorist-3.15p-1: kfreebsd-i386 i386 sparc powerpc
+%libgtkada: kfreebsd-i386 i386 sparc powerpc
+%libgtkada2: kfreebsd-i386 i386 sparc powerpc
+%libopentoken: kfreebsd-i386 i386 sparc powerpc
+%libtexttools2: kfreebsd-i386 i386 sparc powerpc
+%libxmlada1: kfreebsd-i386 i386 sparc powerpc
+%topal: kfreebsd-i386 i386 sparc powerpc
 
 # Depend on gnat-3.3, gnat-3.4
 %music123: !arm !m68k
 ghdl: !m68k
 
 # ghc6 requires itself to build itself
-%ghc6: alpha amd64 hppa i386 m68k s390 sparc ia64 powerpc
-%alex: alpha amd64 hppa i386 m68k s390 sparc ia64 powerpc
-happy: alpha amd64 hppa i386 m68k s390 sparc ia64 powerpc
-haddock: alpha amd64 hppa i386 m68k s390 sparc ia64 powerpc
-%ghc-cvs: alpha amd64 hppa i386 m68k s390 sparc ia64 powerpc	      # changed per Ian Lynagh
+%ghc6: alpha amd64 hppa kfreebsd-i386 i386 m68k s390 sparc ia64 powerpc
+%alex: alpha amd64 hppa kfreebsd-i386 i386 m68k s390 sparc ia64 powerpc
+happy: alpha amd64 hppa kfreebsd-i386 i386 m68k s390 sparc ia64 powerpc
+haddock: alpha amd64 hppa kfreebsd-i386 i386 m68k s390 sparc ia64 powerpc
+%ghc-cvs: alpha amd64 hppa kfreebsd-i386 i386 m68k s390 sparc ia64 powerpc	      # changed per Ian Lynagh
 
 # IEEE-1394 Firewire
 coriander: !m68k
@@ -905,33 +905,33 @@
 
 # dietlibc
 %dietlibc: alpha amd64 arm hppa i386 ia64 mips mipsel powerpc sparc s390    # ANAIS
-fgetty: !m68k							# build-depend on dietlibc-dev:
-fnord: !m68k							# build-depend on dietlibc-dev:
-%kudzu: !m68k							# build-depend on dietlibc-dev:
-%libowfat: !m68k						# build-depend on dietlibc-dev:
-minit: !m68k							# build-depend on dietlibc-dev:
-slidentd: !m68k							# build-depend on dietlibc-dev:
-util-vserver: !m68k						# build-depend on dietlibc-dev:
+fgetty: !m68k !kfreebsd-i386							# build-depend on dietlibc-dev:
+fnord: !m68k !kfreebsd-i386							# build-depend on dietlibc-dev:
+%kudzu: !m68k !kfreebsd-i386							# build-depend on dietlibc-dev:
+%libowfat: !m68k !kfreebsd-i386						# build-depend on dietlibc-dev:
+minit: !m68k !kfreebsd-i386							# build-depend on dietlibc-dev:
+slidentd: !m68k !kfreebsd-i386							# build-depend on dietlibc-dev:
+util-vserver: !m68k !kfreebsd-i386						# build-depend on dietlibc-dev:
 
 # Open office - no 64-bit support until OO.o 2.0
-%openoffice.org: i386 sparc powerpc s390
+%openoffice.org: kfreebsd-i386 i386 sparc powerpc s390
 %openoffice.org-amd64: amd64 ia64
-%openoffice.org1.1: i386 sparc powerpc s390
-%oooqs: i386 sparc powerpc s390					     # Depends on openoffice.org
-%ooqstart: i386 sparc powerpc s390				     # Depends on openoffice.org
+%openoffice.org1.1: kfreebsd-i386 i386 sparc powerpc s390
+%oooqs: kfreebsd-i386 i386 sparc powerpc s390					     # Depends on openoffice.org
+%ooqstart: kfreebsd-i386 i386 sparc powerpc s390				     # Depends on openoffice.org
 
 # mono stuff
-%gtk-sharp: amd64 i386 powerpc arm ia64 s390
-%lasso: amd64 i386 powerpc arm ia64 s390
-%mcs: amd64 i386 powerpc arm ia64 s390
-%mod-mono: amd64 i386                                                  # i386-only, but should it be?
-%mono: amd64 i386 powerpc arm ia64 s390                                     # ANAIS, #259523
-%blam: amd64 i386 powerpc arm ia64 s390
-%f-spot: amd64 i386 powerpc arm ia64 s390
-%libgdiplus: amd64 i386 powerpc arm ia64 s390
-%monodevelop: i386 powerpc arm ia64 s390
-%muine: amd64 i386 powerpc arm ia64 s390
-%prj2make-sharp: amd64 i386 powerpc arm ia64 s390
+%gtk-sharp: amd64 kfreebsd-i386 i386 powerpc arm ia64 s390
+%lasso: amd64 kfreebsd-i386 i386 powerpc arm ia64 s390
+%mcs: amd64 kfreebsd-i386 i386 powerpc arm ia64 s390
+%mod-mono: amd64 kfreebsd-i386 i386                                                  # i386-only, but should it be?
+%mono: amd64 kfreebsd-i386 i386 powerpc arm ia64 s390                                     # ANAIS, #259523
+%blam: amd64 kfreebsd-i386 i386 powerpc arm ia64 s390
+%f-spot: amd64 kfreebsd-i386 i386 powerpc arm ia64 s390
+%libgdiplus: amd64 kfreebsd-i386 i386 powerpc arm ia64 s390
+%monodevelop: kfreebsd-i386 i386 powerpc arm ia64 s390
+%muine: amd64 kfreebsd-i386 i386 powerpc arm ia64 s390
+%prj2make-sharp: amd64 kfreebsd-i386 i386 powerpc arm ia64 s390
 
 # m68k specific
 %quantlib: !m68k						# ROM
@@ -939,6 +939,62 @@
 quantlib-ruby: !m68k
 %rquantlib: !m68k
 
+# Linux-specific
+%aconnectgui: !kfreebsd-i386 !hurd-i386
+%alsa-driver: !kfreebsd-i386 !hurd-i386
+%alsa-lib: !kfreebsd-i386 !hurd-i386
+%alsa-oss: !kfreebsd-i386 !hurd-i386
+%alsa-plugins: !kfreebsd-i386 !hurd-i386
+%alsa-utils: !kfreebsd-i386 !hurd-i386
+%alsamixergui: !kfreebsd-i386 !hurd-i386
+%alsaplayer: !kfreebsd-i386 !hurd-i386
+%clalsadrv: !kfreebsd-i386 !hurd-i386
+%checkpolicy: !kfreebsd-i386 !hurd-i386
+%eagle-usb: !kfreebsd-i386 !i386 !hurd-i386
+%encfs: !kfreebsd-i386 !hurd-i386
+%fireflier: !kfreebsd-i386 !hurd-i386
+%ftm: !kfreebsd-i386 !hurd-i386
+%gnome-alsamixer: !kfreebsd-i386 !i386 !hurd-i386
+%gr-audio-alsa: !kfreebsd-i386 !i386 !hurd-i386
+%hotplug: !kfreebsd-i386 !hurd-i386
+%ieee80211-source: !kfreebsd-i386 !hurd-i386
+%ifenslave-2.4: !kfreebsd-i386 !hurd-i386
+%ifenslave-2.6: !kfreebsd-i386 !hurd-i386
+%ipac-ng: !kfreebsd-i386 !hurd-i386
+%ipchains: !kfreebsd-i386 !hurd-i386
+%iptables: !kfreebsd-i386 !hurd-i386
+kexec-tools: !kfreebsd-i386 !hurd-i386
+%klibc: !kfreebsd-i386 !hurd-i386
+%knetfilter: kfreebsd-i386 i386 !hurd-i386
+%lcrash: !kfreebsd-i386 !hurd-i386
+%libaio: !kfreebsd-i386 !hurd-i386
+%libsemanage: !kfreebsd-i386 !hurd-i386
+%libselinux: !kfreebsd-i386 !hurd-i386
+%linux-2.6: !kfreebsd-i386 !hurd-i386
+%linux-kernel-headers: !kfreebsd-i386 !hurd-i386
+%lksctp-tools: !kfreebsd-i386 !hurd-i386
+%modutils: !kfreebsd-i386 !hurd-i386
+%module-init-tools: !kfreebsd-i386 !hurd-i386
+%murasaki: !kfreebsd-i386 !hurd-i386
+%nufw: !kfreebsd-i386 !hurd-i386
+%paxctl: !kfreebsd-i386 !hurd-i386
+%perlipq: !kfreebsd-i386 !hurd-i386
+%policycoreutils: !kfreebsd-i386 !hurd-i386
+%pwc: !kfreebsd-i386 !hurd-i386
+raidutils: !kfreebsd-i386 !hurd-i386
+%rt2400: !kfreebsd-i386 !hurd-i386
+%rt2500: !kfreebsd-i386 !hurd-i386
+%sfs: !kfreebsd-i386 !hurd-i386
+%shfs: !kfreebsd-i386 !hurd-i386
+%slat: !kfreebsd-i386 !hurd-i386
+%sshfs: !kfreebsd-i386 !hurd-i386
+%dev: !kfreebsd-i386 !hurd-i386
+%usbmgr: !kfreebsd-i386 !hurd-i386
+%uml-utilities: !kfreebsd-i386 !hurd-i386
+wavemon: !kfreebsd-i386 !hurd-i386
+%udev: !kfreebsd-i386 !hurd-i386
+%wireless-tools: !kfreebsd-i386 !hurd-i386
+
 ####################################################################################
 # Contrib and non-free
 # NB: non-free isn't auto-built so this is likely very out-of-date
